### PR TITLE
Remove workaround for inconsistent sv locale collation data file naming

### LINKED ISSUE #2764: ICU 72 doesn't include sv_reformed_meta.toml
Running datagen against the new ICU 72 zip file prints out:

```
WARN  [icu_provider::error] ICU4X data error: Zip: specified file not found in archive
WARN  [icu_provider::error] ICU4X data error: Zip: collation/implicithan/sv_reformed_meta.toml
```

What does it mean? Does something need to be fixed?

CC @echeran @markusicu @hsivonen

### PR DESCRIPTION
**Pull request by echeran**
Fixes #2764

### CODE REVIEW

**Comment by echeran** [comment]
Something is broken when I test locally, so even if this PR is on the right track, it needs fixing somewhere.

Does this PR need to be rebased on changes that points to the `release-72-1` tag of icuexportdata somewhere?

File: provider/testdata/Cargo.toml [hunk 1]
```diff
@@ -166,12 +166,12 @@ icuexportdata_glob = [
     "collation/implicithan/root_standard_jamo.toml",
     "collation/implicithan/root_standard_meta.toml",
     "collation/implicithan/root_standard_prim.toml",
-    "collation/implicithan/sv_reformed_data.toml",
```
Comment (sffc) [inline provider/testdata/Cargo.toml:169]: If you change these paths, you need to re-run `cargo make testdata-download-sources`

**Review by sffc — COMMENTED** [review]
Thanks for looking into this!

All of the CIs should pass.

**Comment by sffc** [comment]
> Does this PR need to be rebased on changes that points to the `release-72-1` tag of icuexportdata somewhere?

You should change `icuexportdata_gitref` in testdata/Cargo.toml

**Comment by hsivonen** [comment]
> Also, we should continue working with ICU 71 data, I think, since that is what we shipped 1.0 with.

Yeah, it seems to me that we don't need to be in a hurry with removing this workaround. For now, datagen should work with CLDR before and after the change to Swedish naming.

**Comment by sffc** [comment]
Discussion: we should mark this code as "deprecated" or "to be removed in 2.0" (maybe with a feature or cfg?), and clean up the warnings.

**Comment by robertbastian** [comment]
As discussed in the meeting, this removes the workaround and breaks with pre-72 data. We can make this breaking change in 2.0. https://github.com/unicode-org/icu4x/issues/2856

**Pull request closed without merging**